cmd/flags.go

```diff
@@ -68,23 +73,39 @@ func ISPProvider() isp.Provider {
 	return ispProvider
 }
 
-func RedisClient() *redis.Client {
-	if *RedisAddr == "" {
-		log.Debug("Redis not configured")
-		return nil
+// RedisClient creates a new redis client.
+func RedisClient() *rclient.Client {
+	opts := &rclient.Options{
+		PoolSize:     10,
+		ReadTimeout:  1 * time.Minute,
+		WriteTimeout: 1 * time.Minute,
+		IdleTimeout:  4 * time.Minute,
+		Password:     *redisPassword,
+		Addr:         *redisAddr,
+		DialTimeout:  30 * time.Second,
 	}
-	log.Debugf("Connecting to Redis at %v", *RedisAddr)
-	redisClient, err := tlsredis.GetClient(&tlsredis.Options{
-		RedisURL:       *RedisAddr,
-		RedisCAFile:    *RedisCA,
-		ClientPKFile:   *RedisClientPK,
-		ClientCertFile: *RedisClientCert,
-	})
-	if err == nil {
-		log.Debugf("Connected to Redis at %v", *RedisAddr)
-		return redisClient
+
+	var cert []byte
+	if _, err := os.Stat(*redisCA); err != nil {
+		cert = []byte(*redisCA)
 	} else {
-		log.Errorf("Unable to connect to redis: %v", err)
+		cert, err = ioutil.ReadFile(*redisCA)
+		if err != nil {
+			log.Debugf("Could not read CA file at %v", *redisCA)
+			return nil
+		}
+	}
+
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(cert)
+	opts.TLSConfig = &tls.Config{
+		RootCAs: certPool,
+	}
+	client := rclient.NewClient(opts)
+
+	if e := client.Ping().Err(); e != nil {
```

Comment:
I believe that this is a change from current behavior. Currently, if redis happens to be unavailable at the time that zeno starts, we'll still get a client but it will fail connect until Redis becomes available again. In the new scheme, we'll get a nil client, which will cause our process to not start.

Previously, we got a nil client only if there was a more permanent configuration issue, not a transient network issue.

Reply:
Huh, ok so you would prefer that it just start even though it can't hit redis at that moment correct? Happy to make that change if you like.

Reply:
Yes. The way it works today, zeno can start even if redis is down, and any functionality that doesn't touch redis still works.

http-proxy was designed to work similarly, we should make sure that we don't break that in the transition to the new client.